Classes/Exception/NoSuchHaikuException.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\Exception;
+
+/**
+ * Exception thrown if a haiku record was not found
+ */
+class NoSuchHaikuException extends \InvalidArgumentException
```

Comment:
We should not use InvalidArgumentException as it inherits from LogicException. A LogicException should be thrown when there is a logical error (from the developer). This is more a RuntimeException (as a result is not as expected due to non-existing uid at runtime), so we should inherit from that. Maybe directly from RuntimeException or more specific UnexpectedValueException. I tend to RuntimeException.